src/Extensions/Logging/Internal/OmexLogsEventSource.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	[EventSource(Name = "Microsoft-OMEX-Logs")]
+	internal sealed class OmexLogsEventSource : EventSource
+	{
+		static OmexLogsEventSource()
+		{
+			Process process = Process.GetCurrentProcess();
+			ProcessName = string.Format("{0} (0x{1:X4})", process.ProcessName, process.Id);
+		}
+
+
+		private static string ProcessName { get; }
+
+
+		public OmexLogsEventSource(IMachineInformation machineInformation, IServiceContext context)
+		{
+			m_machineInformation = machineInformation;
+			m_serviceContext = context;
+		}
+
+
+		private readonly IMachineInformation m_machineInformation;
+		private readonly IServiceContext m_serviceContext;
+
+		private string ApplicationName => m_machineInformation.MachineRole;
+		private string ServiceName => m_machineInformation.ServiceName;
+		private string AgentName => m_machineInformation.AgentName;
+		private string BuildVersion => m_machineInformation.BuildVersion;
+
+		private Guid PartitionId => m_serviceContext?.PartitionId ?? Guid.Empty;
+		private long ReplicaId => m_serviceContext?.ReplicaOrInstanceId ?? 0;
+
+		private string CorrelationId => Activity.Current?.Id ?? string.Empty; //Breaking Change: Type Changed from Guid ?? Guid.Empty
```

Comment:
>//Breaking Change: Type Changed from Guid ?? Guid.Empty [](start = 72, length = 55)

can we track as an Issue to remove the breaking change comments once this is integrated

Reply:
I would like to resolve them before merging, but I'll need @vlad-ion help for it, if I won't be able to do it I'll create an issue

Reply:
Issue created https://github.com/microsoft/Omex/issues/162